java/com/google/cose/utils/CborUtils.java

```diff
@@ -0,0 +1,154 @@
+ */
+public class CborUtils {
+
+  /**
+   * Decodes cbor byte encoding into a CBOR data item.
+   * @param data byte array in cbor format.
+   * @return DataItem cbor object
+   */
+  public static DataItem decode(final byte[] data) {
+    final ByteArrayInputStream bais = new ByteArrayInputStream(data);
+    try {
+      final List<DataItem> dataItems = new CborDecoder(bais).decode();
+      if (dataItems.size() != 1) {
+        throw new CoseException("Byte stream cannot be decoded properly. Expected 1 item, found "
+            + dataItems.size());
+      }
+      return dataItems.get(0);
+    } catch (final CborException ex) {
+      throw new CoseException("Error decoding data", ex);
+    }
+  }
+
+  /**
+   * Converts CBOR data item into byte encoding.
+   * @param dataItem DataItem cbor object
+   * @return encoded bytes
+   */
+  public static byte[] encode(final DataItem dataItem) {
+    final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    CborEncoder encoder = new CborEncoder(baos);
+    try {
+      encoder.encode(dataItem);
+    } catch (final CborException ex) {
+      throw new CoseException("Error encoding data", ex);
+    }
+    return baos.toByteArray();
+  }
+
+  /**
+   * Returns the {@link DataItem} as a {@link Map}.
+   * @param dataItem cborObject to be converted to Map.
+   * @return Map object
+   */
+  public static Map asMap(final DataItem dataItem) throws CborException {
+    if (dataItem.getMajorType() != MajorType.MAP) {
+      throw new CborException(
+          String.format("Expected a map, got %s", dataItem.getMajorType().name()));
+    }
+    return (Map) dataItem;
+  }
+
+  /**
+   * Returns the {@link DataItem} as an {@link Array}.
+   * @param dataItem cborObject to be converted to Array.
+   * @return Array object
+   */
+  public static Array asArray(final DataItem dataItem) throws CborException {
+    if (dataItem.getMajorType() != MajorType.ARRAY) {
+      throw new CborException(
+          String.format("Expected an array, got %s", dataItem.getMajorType().name()));
+    }
+    return (Array) dataItem;
+  }
+
+  /**
+   * Returns the {@link DataItem} as a {@link ByteString}.
+   * @param dataItem cborObject to be converted to ByteString.
+   * @return ByteString object
+   */
+  public static ByteString asByteString(final DataItem dataItem) throws CborException {
+    if (dataItem.getMajorType() != MajorType.BYTE_STRING) {
+      throw new CborException(
+          String.format("Expected a byte string, got %s", dataItem.getMajorType().name()));
+    }
+    return (ByteString) dataItem;
+  }
+
+  /**
+   * Returns DataItem from a cbor map based on Integer index.
+   * @param cborMap map that has the information.
+   * @param index integer index to be used as key in the map.
+   * @return value in the map corresponding to key
+   */
+  public static DataItem getValueFromMap(final Map cborMap, final int index) {
```

Comment:
Unused?

Reply:
yeah. This is being used for Key constructors. I can remove this for now and add it again later.